R/rdeephaven/R/client_wrapper.R

```diff
@@ -1,110 +1,249 @@
-#' client_options <- ClientOptions$new()
-#' client <- Client$new(target = "localhost:10000", client_options = client_options)
-#'
-#' # open a table that already exists on the server
-#' new_table_handle1 <- client$open_table("table_on_the_server")
-#'
-#' # create a new dataframe, import onto the server, and retrieve a reference
-#' new_data_frame <- data.frame(matrix(rnorm(10 * 1000), nrow = 10))
-#' new_table_handle2 <- client$import_table(new_data_frame)
-#'
-#' # run a python script on the server (default client options specify a Python console)
-#' client$run_script("print([i for i in range(10)])")
-Client <- R6Class("Client",
-  public = list(
-
-    #' @description
-    #' Connect to a running Deephaven server.
-    #' @param target The address of the Deephaven server.
-    #' @param client_options ClientOptions instance with the parameters needed to connect to the server.
-    #' See ?ClientOptions for more information.
-    initialize = function(target, client_options) {
-      .verify_string("target", target)
-      if (class(client_options)[[1]] != "ClientOptions") {
-        stop(paste("'client_options' should be a Deephaven ClientOptions object. Got an object of type", class(client_options)[[1]], "instead."))
-      }
-      private$internal_client <- new(INTERNAL_Client,
-        target = target,
-        client_options = client_options$internal_client_options
-      )
-    },
-
-    #' @description
-    #' Opens a table named 'name' from the server if it exists.
-    #' @param name Name of the table to open from the server, passed as a string.
-    #' @return TableHandle reference to the requested table.
-    open_table = function(name) {
-      .verify_string("name", name)
-      if (!private$check_for_table(name)) {
-        stop(paste0("The table '", name, "' you're trying to pull does not exist on the server."))
-      }
-      return(TableHandle$new(private$internal_client$open_table(name)))
-    },
-
-    #' @description
-    #' Imports a new table to the Deephaven server. Note that this new table is not automatically bound to
-    #' a variable name on the server. See `?TableHandle` for more information.
-    #' @param table_object An R Data Frame, a dplyr Tibble, an Arrow Table, or an Arrow RecordBatchReader
-    #' containing the data to import to the server.
-    #' @return TableHandle reference to the new table.
-    import_table = function(table_object) {
-      table_object_class <- class(table_object)
-      if (table_object_class[[1]] == "data.frame") {
-        return(TableHandle$new(private$df_to_dh_table(table_object)))
+#' @export
+setClass(
+  "Client",
+  representation(
+    .internal_rcpp_object = "Rcpp_INTERNAL_Client"
+  )
+)
+
+setGeneric(
+  "connect",
+  function(target, ...) {
+    return(standardGeneric("connect"))
+  },
+  signature = c("target")
+)
+
+#' @export
+setMethod(
+  "connect",
+  signature = c(target = "character"),
+  function(target,
+           auth_type = "anonymous",
```

Comment:
My understanding of `signature` is that it gives the argument names and types of the arguments to be dispatched on. Since `auth_type` must be a string, adding `signature = c(target = "character", auth_type = "character")` does not help dispatch, if we were to use it.

If you mean that `auth_type` should not have a default and must be user-specified, that is a different question. I think anon default makes sense.